packages/p/protobuf-cpp/xmake.lua

```diff
@@ -38,15 +38,26 @@ package("protobuf-cpp")
         if package:config("zlib") then
             package:add("deps", "zlib")
         end
+        if package:version():gt("3.19.4") then
+            package:add("deps", "abseil")
+        end
     end)
 
     on_install("windows", "linux", "macosx", function (package)
-        -- os.cd("cmake")
+        if package:version():le("3.19.4") then
+            os.cd("cmake")
+        end
         io.replace("CMakeLists.txt", "set(protobuf_DEBUG_POSTFIX \"d\"", "set(protobuf_DEBUG_POSTFIX \"\"", {plain = true})
-        local configs = {"-Dprotobuf_BUILD_TESTS=OFF",
-                         "-Dprotobuf_BUILD_PROTOC_BINARIES=ON",
-                         "-Dprotobuf_ABSL_PROVIDER=package"}
+     
+        local configs = {"-Dprotobuf_BUILD_TESTS=OFF", "-Dprotobuf_BUILD_PROTOC_BINARIES=ON"}
         table.insert(configs, "-DBUILD_SHARED_LIBS=" .. (package:config("shared") and "ON" or "OFF"))
+      
+        local packagedeps = {}
+        if package:version():gt("3.19.4") then
+            table.insert(packagedeps, "abseil")
```

Comment:
if abseil is optional, we can provide a abseil config and disable it to ensure minimal dependencies by default. 

like zlib. https://github.com/xmake-io/xmake-repo/blob/6f45709808d41ebfd1aa17b7757622d44416021d/packages/p/protobuf-cpp/xmake.lua#L22

Reply:
for that, it's interesting that abseil is configured with c++17 at https://github.com/xmake-io/xmake-repo/blob/090ff77b283502610fce37a91c6baff01339f785/packages/a/abseil/xmake.lua#L35 instead of 14 listed in https://github.com/abseil/abseil-cpp/blob/master/CMake/README.md

Reply:
> for that, it's interesting that abseil is configured with c++17 at
> 
> https://github.com/xmake-io/xmake-repo/blob/090ff77b283502610fce37a91c6baff01339f785/packages/a/abseil/xmake.lua#L35
> 
> instead of 14 listed in https://github.com/abseil/abseil-cpp/blob/master/CMake/README.md

it might be possible to set the C++ version of abseil to C++14 in the repo, I opened up another PR to see: https://github.com/xmake-io/xmake-repo/pull/3681